source/lib/vagrant-openstack-provider/version_checker.rb

```diff
@@ -0,0 +1,74 @@
+require 'colorize'
+require 'singleton'
+require 'vagrant-openstack-provider/version'
+
+module VagrantPlugins
+  module Openstack
+    class VersionChecker
+      include Singleton
+
+      #
+      # :latest, :outdated or :unstable
+      #
+      # A version is considered unstable if it does not
+      # respect the pattern or if it is greater than the
+      # latest from rubygem
+      #
+      attr_accessor :status
+
+      def initialize
+        @status = nil
+      end
+
+      #
+      # Check the latest version from rubygem and set the status
+      #
+      def check
```

Comment:
I agree for the timeout. The global rescue is done in the module method who wrap this call.